builtin/providers/aws/validators.go

```diff
@@ -353,7 +353,7 @@ func validateArn(v interface{}, k string) (ws []string, errors []error) {
 	}
 
 	// http://docs.aws.amazon.com/lambda/latest/dg/API_AddPermission.html
-	pattern := `^arn:[\w-]+:([a-zA-Z0-9\-])+:([a-z]{2}-[a-z]+-\d{1})?:(\d{12})?:(.*)$`
+	pattern := `^arn:[\w-]+:([a-zA-Z0-9\-])+:([a-z]{2}-(gov-)?[a-z]+-\d{1})?:(\d{12})?:(.*)$`
```

Comment:
As you can imagine - I'm fresh on these docs 😄 

The second `partition` field changes in GovCloud and China (Beijing), but we're already loose about the contents of that field. China's `region` identifier follows the "normal" format - `cn-north-1`.

In other words, we should be fine for China!

I'd love to see ARN parsing and validating not be up to a regexp, but for now, 🤷‍♂️